api/pkg/db/migration/202107291608_refresh_all_tables.go

```diff
@@ -0,0 +1,274 @@
+
+package migration
+
+import (
+	"github.com/go-gormigrate/gormigrate/v2"
+	"github.com/tektoncd/hub/api/gen/log"
+	"github.com/tektoncd/hub/api/pkg/db/model"
+	"gorm.io/gorm"
+)
+
+func refreshAllTables(log *log.Logger) *gormigrate.Migration {
+	return &gormigrate.Migration{
+		ID: "202107291608_refresh_all_tables",
+		Migrate: func(db *gorm.DB) error {
+			txn := db.Begin()
+			err := migrateDB(txn, log)
+			if err != nil {
+				txn.Rollback()
+				log.Error(err)
+			}
+			txn.Commit()
+			return nil
+		},
+	}
+}
+
+func migrateDB(txn *gorm.DB, log *log.Logger) error {
+	var user []model.User
+	if err := txn.Find(&user).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var user_scopes []model.UserScope
+	if err := txn.Find(&user_scopes).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var user_rating []model.UserResourceRating
+	if err := txn.Find(&user_rating).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var categories []model.Category
+	if err := txn.Find(&categories).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var tags []model.Tag
+	if err := txn.Find(&tags).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var sync_job []model.SyncJob
+	if err := txn.Find(&sync_job).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var scopes []model.Scope
+	if err := txn.Find(&scopes).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var resources []model.Resource
+	if err := txn.Find(&resources).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var resource_versions []model.ResourceVersion
+	if err := txn.Find(&resource_versions).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var resource_tags []model.ResourceTag
+	if err := txn.Find(&resource_tags).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var configs []model.Config
+	if err := txn.Find(&configs).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var catalog []model.Catalog
+	if err := txn.Find(&catalog).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	var catalog_error []model.CatalogError
+	if err := txn.Find(&catalog_error).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	if err := txn.Migrator().DropTable(
+		&model.Category{},
+		&model.Tag{},
+		&model.Catalog{},
+		&model.CatalogError{},
+		&model.Resource{},
+		&model.ResourceVersion{},
+		&model.User{},
+		&model.UserResourceRating{},
+		&model.SyncJob{},
+		&model.Scope{},
+		&model.UserScope{},
+		&model.Config{},
+		&model.ResourceTag{},
+	); err != nil {
+		log.Error(err)
+		return err
+	}
+
+	if err := txn.AutoMigrate(
+		&model.Category{},
+		&model.Tag{},
+		&model.Catalog{},
+		&model.CatalogError{},
+		&model.Resource{},
+		&model.ResourceVersion{},
+		&model.User{},
+		&model.UserResourceRating{},
+		&model.SyncJob{},
+		&model.Scope{},
+		&model.UserScope{},
+		&model.Config{},
+		&model.ResourceTag{},
+	); err != nil {
+		log.Error(err)
+		return err
+	}
+
+	if len(catalog) > 0 {
```

Comment:
just curious when we fetch records and insert as it is ... do the creation timestamp changes ? or remains same as before

Reply:
I think it should remain the same since we are inserting the data which we have collected from the db itself and also the id for all is not changed so I think it should remain the same